src/utils.jl

```diff
@@ -19,6 +19,65 @@ cholesky_factor(X::Cholesky) = X.U
 cholesky_factor(X::UpperTriangular) = X
 cholesky_factor(X::LowerTriangular) = X
 
+# TODO: Add `check` as an argument?
+"""
+    cholesky_lower(X)
+
+Return the lower triangular Cholesky factor of `X` as a `Matrix`
+rather than `LowerTriangular`.
+
+!!! note
+    This is a thin wrapper around `cholesky(Hermitian(X)).L`
+    but with a custom `ChainRulesCore.rrule` implementation.
+"""
+cholesky_lower(X::AbstractMatrix) = lower_triangular(parent(cholesky(Hermitian(X)).L))
```

Comment:
I wrap in `Hermitian` to effectively do the same as the current implementation of `cholesky_factor` but I believe `cholesky(::Hermitian)` is only valid starting from Julia 1.8 (going by a comment in BijectorsReverseDiffExt), so we need to fix this.

Reply:
This is actually not a problem anymore since we're now defining the adjoint to circumvent the `cholesky` on tracked completely.